### PR TITLE
Aliwoto dev

### PR DESCRIPTION
**Pull request by ALiwoto**
added the sticker event parser, 
also forward parameters and new chat parameters.

### CODE REVIEW

File: src/gwbot/include/gwbot/lib/tg_event/text.h [hunk 1]
```diff
@@ -17,12 +17,17 @@
...
 	struct tgevi_chat	chat;
 	time_t			date;
-	const char		*text;
+        time_t			forward_date;
+	const char              *text;
 	uint16_t		entity_c; /* How many entities[n] there are? */
 	struct tgevi_entity	*entities;
 	struct tgev		*reply_to;
+        bool			is_forwarded;
+	bool			is_unknown_fwd;
```
Comment (ammarfaizi2) [inline src/gwbot/include/gwbot/lib/tg_event/text.h:30]: Bad indentation, and don't use space for indentation.

File: tests/006_tg_event/tg_event_photo.c [hunk 2]
```diff
@@ -110,7 +216,7 @@ static int test_tg_event_photo_001_simple(void)
 
 
 struct list_func tg_event_photo_list[] = {
-	{test_tg_event_photo_001_simple, 220},
+	{test_tg_event_photo_002_simple, 220},
 	{NULL, 0}
```
Comment (ammarfaizi2) [inline tests/006_tg_event/tg_event_photo.c:220]: Don't remove `test_tg_event_photo_001_simple` from the list, just add more.

File: tests/006_tg_event/tg_event_text.c [hunk 3]
```diff
@@ -65,8 +65,70 @@ static int test_tg_event_text_001_simple(void)
...
+
+	tg_event_destroy(&evt);
+	return 0;
+}
+
+
 struct list_func tg_event_text_list[] = {
-	{test_tg_event_text_001_simple, 230},
+	{test_tg_event_text_002_simple, 150},
 	{NULL, 0}
```
Comment (ammarfaizi2) [inline tests/006_tg_event/tg_event_text.c:132]: Don't remove `test_tg_event_text_002_simple` from the list, just add more.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
Please don't remove the finished tests from the list. You removed 001_simple for text and photo. Please fix it before I merge this.

File: tests/006_tg_event/tg_event_text.c (hunk 3 above)
Comment (ALiwoto) [reply tests/006_tg_event/tg_event_text.c:132]: okay, I'll fix them right now.

**Comment by ammarfaizi2** [comment]
Are you sure you fixed it?
You clearly didn't test-compiled it and said everything is fixed, did you?

```
ammarfaizi2@integral:~/project/now/GNUWeebBot$ make test
--- Build debug mode
[====] Synthesis: Tested: 3 | Passing: 3 | Failing: 0 | Crashing: 0 
[====] Synthesis: Tested: 5 | Passing: 5 | Failing: 0 | Crashing: 0 
[====] Synthesis: Tested: 6 | Passing: 6 | Failing: 0 | Crashing: 0 
   CC		 tests/006_tg_event/tg_event_text.o
/home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/tg_event_text.c:132:2: error: expected ‘}’ before ‘{’ token
  132 |  {test_tg_event_text_002_simple, 150},
      |  ^
/home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/tg_event_text.c:130:41: note: to match this ‘{’
  130 | struct list_func tg_event_text_list[] = {
      |                                         ^
/home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/tg_event_text.c:68:12: warning: ‘test_tg_event_text_002_simple’ defined but not used [-Wunused-function]
   68 | static int test_tg_event_text_002_simple(void)
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[1]: *** [/home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/Makefile:31: /home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/tg_event_text.o] Error 1
make: *** [/home/ammarfaizi2/project/now/GNUWeebBot/tests/Makefile:21: test] Error 2
ammarfaizi2@integral:~/project/now/GNUWeebBot$ 
```

**Comment by ammarfaizi2** [comment]
Merged, thanks.

**Comment by ALiwoto** [comment]
let's go for next part
what's the next step??

**Comment by ammarfaizi2** [comment]
I will fix the unit test first, wait for it. We will be working with video and voice event.

**Comment by ALiwoto** [comment]
@ammarfaizi2 nice, good luck
and I'm looking forward for it

**Pull request merged**